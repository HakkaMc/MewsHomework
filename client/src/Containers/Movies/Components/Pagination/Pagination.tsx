import React from "react";
import { connect } from "react-redux";
import PaginationComponent from "@material-ui/lab/Pagination";
import { compose } from "recompose";
import classnames from "classnames";

import { ReduxState, actions } from "../../../../reducers";

export interface Props {
	searchMovies: typeof actions.movies.searchMovies;
	page: number;
	totalPages: number;
	className?: string;
}

export const Pagination: React.FC<Props> = (props: Props) => {
	const changePage = (event: any, pageNumber: number) =>
		props.searchMovies(undefined, pageNumber);

	if (!props.totalPages) return null;

	return (
		<div className={classnames(props.className)}>
			<PaginationComponent
				count={props.totalPages}
				page={props.page}
				onChange={changePage}
			/>
		</div>
	);
};

export default compose<Props, any>(
	connect(
		(state: ReduxState) => ({
			page: state.movies.page,
			totalPages: state.movies.totalPages
		}),
		{
			searchMovies: actions.movies.searchMovies
		}
	)
)(Pagination);
