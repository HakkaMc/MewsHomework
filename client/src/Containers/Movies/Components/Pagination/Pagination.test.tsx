import React from "react";
import { shallow } from "enzyme";
import PaginationComponent from "@material-ui/lab/Pagination";

import { Pagination, Props } from "./Pagination";

describe("<Pagination/>", () => {
	let props: Props;

	beforeEach(() => {
		props = {
			page: 1,
			searchMovies: jest.fn(),
			totalPages: 1
		};
	});

	it("Match snapshot with totalPages=0", () => {
		props.totalPages = 0;
		const wrapper = shallow(<Pagination {...props} />);
		expect(wrapper.isEmptyRender()).toBe(true);
		expect(wrapper).toMatchSnapshot();
	});

	it("Match snapshot with totalPages=1", () => {
		props.totalPages = 1;
		const wrapper = shallow(<Pagination {...props} />);
		expect(wrapper.isEmptyRender()).toBe(false);
		expect(wrapper).toMatchSnapshot();
	});

	it("Click PaginationComponent calls searchMovies function", () => {
		props.totalPages = 1;
		const wrapper = shallow(<Pagination {...props} />);
		// @ts-ignore
		wrapper.find(PaginationComponent).prop("onChange")(null, 1);
		expect(props.searchMovies).toHaveBeenCalled();
	});
});
