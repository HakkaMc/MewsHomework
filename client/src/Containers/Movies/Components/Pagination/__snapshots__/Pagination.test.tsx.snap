// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> Match snapshot with totalPages=0 1`] = `""`;

exports[`<Pagination/> Match snapshot with totalPages=1 1`] = `
<div
  className=""
>
  <WithStyles(ForwardRef(Pagination))
    count={1}
    onChange={[Function]}
    page={1}
  />
</div>
`;
